data-pipeline/src/pipeline/main.py

```diff
@@ -441,20 +463,121 @@ def run_compute_rag(set_type, year):
     return time_taken
 
 
-def handle_msg(worker_client, msg, worker_queue, complete_queue):
-    msg_payload = json.loads(msg.content)
-    try:
-        msg_payload["pre_process_duration"] = pre_process_data(
-            worker_client, msg_payload["type"], msg_payload["year"]
+def run_user_defined_rag(
+    set_type: str,
+    year: int,
+    run_id: str,
+    comparator_set: pd.DataFrame,
+):
+    """
+    Perform user-defined RAG calculations.
+
+    Use the pre-processed "all-schools" data to guarantee coverage of
+    the user-defined comparator-set.
+
+    :param set_type: the triggering message type
+    :param year: financial year in question
+    :param run_id: unique run identifier
+    :param comparator_set: user-defined comparator-set
+    :return: duration of RAG calculations
+    """
+    all_schools = prepare_data(
+        pd.read_parquet(
+            get_blob("pre-processed", f"{set_type}/{year}/all_schools.parquet")
         )
+    )
 
-        msg_payload["comparator_set_duration"] = compute_comparator_sets(
-            msg_payload["type"], msg_payload["year"]
-        )
+    st = time.time()
+    logger.info(f"Computing user-defined RAG ({run_id}).")
+    df = pd.DataFrame(compute_user_defined_rag(all_schools, comparator_set)).set_index(
+        "URN"
+    )
+    logger.info(
+        f"Computing user-defined ({run_id}) RAG. Done in {time.time() - st:.2f} seconds."
+    )
 
-        msg_payload["rag_duration"] = run_compute_rag(
-            msg_payload["type"], msg_payload["year"]
-        )
+    write_blob(
+        "metric-rag",
+        f"{set_type}/{run_id}/user-defined.parquet",
+        df.to_parquet(),
+    )
+
+    insert_metric_rag(
```

Comment:
The `run_type`/`set_type` variables in `insert_metric_rag()` aren't entirely clear—is this right? Currently `RunId` ends up populated with `year` so this seemed to make sense…?

Reply:
Yep... this is correct, just for clarity you are correct the naming around these is shocking *apologies* but 

* `set_type` - repersents the type of run `default`, `custom_comparator` or `custom_data`
* `run_type` - represents whether the set being computed, is Mixed (Maintained Schools and Academies) or Unmixed separate
* `run_id` - In the default case == Year, but for this it is a FE supplied ID, and the year is assumed to be the current year @jrabbott  would be best to confirm this concretely thou.

Reply:
Updated and documented as per discussions here and on Slack.